src/snapshot_tests.rs

```diff
@@ -12,22 +16,66 @@ mod utils;
 
 #[test]
 fn test_snapshots() {
-    if env::var("CI").is_ok() {
-        return;
+    let artifact_save_path = artifacts_path();
+    if artifact_save_path.exists() {
+        fs::remove_dir_all(artifact_save_path).unwrap();
     }
 
-    let mut produced_snapshots = HashSet::new();
-    for snapshot_test in snapshot_tests() {
-        eprintln!("Test \"{}\"", snapshot_test.name);
-        let snapshots = snapshot_test.generate_snapshots().unwrap();
-        match snapshot_test.test_snapshots(&snapshots) {
-            Ok(_) => produced_snapshots.extend(snapshots.iter().map(Snapshot::save_path)),
-            Err(err) => panic!("{err}"),
+    let test_cases = snapshot_tests();
+    for test_case in test_cases.iter() {
+        eprintln!("Test \"{}\"", test_case.name);
+        let snapshots = test_case.generate_snapshots().unwrap();
+        if let Err(err) = test_case.test_snapshots(&snapshots) {
+            handle_error(err);
         }
     }
 
-    let unused_snapshots = find_unused_snapshots(&produced_snapshots, snapshots_path());
-    if !unused_snapshots.is_empty() {
-        panic!("Some snapshots were not used: {unused_snapshots:#?}")
+    // Check for unused snapshots
+    let snapshot_paths = test_cases
+        .iter()
+        .flat_map(TestCase::snapshot_paths)
+        .collect::<HashSet<_>>();
+    for path in find_unused_snapshots(&snapshot_paths, snapshots_path()) {
+        println!("Removed unused snapshot {path:?}");
+        fs::remove_file(path).unwrap();
+    }
+}
+
+fn handle_error(err: Error) {
+    let Some(test_case_err) = err.downcast_ref::<TestCaseError>() else {
+        panic!("{err}");
+    };
```

Comment:
```suggestion
    let test_case_err = err.downcast_ref::<TestCaseError>().unwrap();
```
also I would prefer to avoid that downcast, but given it's a test I'm fine with leaving it if it's problematic